### DISCUSSION TITLE
Datagrid Delete Popup

### DISCUSSION THREAD

**Discussion by peterhym21** [Megabit/Blazorise#3860] [Q&A] [answered]
Is there any way to get a Popup on the delete button to confirm that you want to delete that item, Directly in Datagrid or do u have to use a Modal for that

                            <DataGrid TItem="UiOperationListVm"
                                  Data="@Operations"
                                  @bind-SelectedRow="@SelectedOperation"
                                  EditMode="DataGridEditMode.Inline"
                                  Editable
                                  PageSize="10"
                                  ShowPager
                                  Responsive>

                                <DataGridColumn Field="@nameof(UiOperationListVm.Id)" Caption="ID" Sortable="false" Displayable="true" />
                                <DataGridColumn Field="@nameof(UiOperationListVm.Name)" Caption="Name" Editable />
                                <DataGridColumn Field="@nameof(UiOperationListVm.Description)" Caption="Description" Editable />
                                <DataGridColumn Field="OperationType.Description" Caption="Operation types" Editable>
                                    <EditTemplate>
                                        <SelectList TItem="UiOperationTypeListVm"
                                                TValue="int"
                                                Data="@OperationTypeListVm"
                                                TextField="@((item)=>item.Description)"
                                                ValueField="@((item)=>(int)item.Name)"
                                                SelectedValue="@UpdateOperationCommand.OperationTypeName"
                                                SelectedValueChanged="@MyListValueChangedHandlerUpdate"
                                                DefaultItemText="select a Operation Type"
                                                class="btn btn-info dropdown-toggle" />
                                    </EditTemplate>
                                </DataGridColumn>
                                <DataGridCommandColumn Width="100px">
                                    <EditCommandTemplate>
                                        <Button Color="Color.Primary" Clicked="@context.Clicked">Edit</Button>
                                    </EditCommandTemplate>
                                    <DeleteCommandTemplate>
                                        <Button Color="Color.Primary" Clicked="@(()=>OnDeletClickedOperation(context))">@context.LocalizationString</Button>
                                    // popup with the confirm delete
                                    </DeleteCommandTemplate>
                                    <SaveCommandTemplate>
                                        <Button Color="Color.Primary" Clicked="@(()=>OnSaveClickedOperation(context))">@context.LocalizationString</Button>
                                    </SaveCommandTemplate>
                                    <CancelCommandTemplate><Button Color="Color.Danger" Clicked="@context.Clicked">Cancel</Button></CancelCommandTemplate>
                                    <NewCommandTemplate></NewCommandTemplate>

                                </DataGridCommandColumn>
                            </DataGrid>

**Comment by stsrki** [comment, marked as answer]
You can call the message service from the OnDeletClickedOperation method. See an example on https://blazorise.com/docs/services/message
